proposals/xxxx-package-manager-performance-testing.md

```diff
@@ -0,0 +1,63 @@
+# Package Manager Performance Testing Conventions
+
+* Proposal: [SE-XXXX](xxx-package-manager-performance-testing.md)
+* Author: [Ankit Aggarwal](https://github.com/aciidb0mb3r)
+* Review Manager: TBD
+* Status: Discussion
+
+## Introduction
+
+This is a proposal for adding package manager support for building and running performance tests in release mode.
+
+## Motivation
+
+The package manager supports building and running tests using the `swift test` subtool. The sources are built in debug mode with swift compiler's
+`-enable-testing` flag allowing testable imports in test modules (which gives it internal level access to the imported module). 
+This works well for normal unit tests, however for performance tests we are usually interested in release build test results which takes advantages of compiler 
+optimizations.
+
+We can provide an option to compile and run the tests in release mode with `-enable-testing` flag. However that will disable some of the compiler optimizations, which 
+may be of interest to packages having very performance sensitive code (which critically depends on those optimization).
+
+Since performance tests usually take more time to run than other tests, there is also a requirement to have a way to run performance tests separate from unit tests.
+As a concrete example, package manager's own performance tests are disabled from running on Swift CI.
```

Comment:
As a further point on motivation, we should explain that we are prevented from doing this currently because we cannot determine which tests do and do not have performance tests.

We also need to note somewhere that although XCTest allows including performance tests anywhere, we are designing particularly workflows based on the idea of users mostly segregating their performance tests into their own module, to allow us to manage the "testability" flag appropriately.